Added the notation *(*) to select all atoms except H in restraints

pascal22p opened the pull request.

richardicooper commented:
Build failures are due to a dependency not downloading - hopefully temporary problem.

richardicooper commented:
Build and test is now OK. Is the 'do not merge' label related to this, or something else?

richardicooper commented:
Code changes look fine.

pascal22p commented:
Something else. I need more tests.

richardicooper commented:
It looks like the new code doesn't warn on non-existent atoms anymore - the failing diff is in lexical.tst output - the old code prints a warning about C(128). The line looks strange though - what's the meaning of the next atom C(2,128)?  

Comparing files lexical.out and MIN64GHGUI.ORG/LEXICAL.OUT
***** lexical.out
#LIST 16
#END
***** MIN64GHGUI.ORG/LEXICAL.OUT
#LIST 16
E Warning: UQISO 0.01 C(128) C(2, 128)
E ------------------^^^^^^^
E Warning: atom C(128) is not present in the model
#END
*****

pascal22p commented:
I fixed the warning on non-existing atom.
Also fixed a bug on symmetry operator check
Introduce a new bug I need to fix. comma is at the wrong place...

pascal22p commented:
The piece of work left is this `*(...)` syntax to select atom where the * is any type.
Unfortunately * is also the sign for the multiplication....

I can work out most of the cases except this one:
*(3) which could be either:
- times 3
- any atom type with a serial of 3

The current syntax analyser would probably interpret this as a valid multiplication so will keep the current behaviour.

The rest is:
1. *(*): everything
2. *(*, part=2): all atoms in part 2
3. *(*, resi=3): all atoms in residue 3

I wondering if 2. and 3. is already possible. I have not checked yet.

pascal22p commented:
rebased branch with master
